### PR TITLE
Support modules with the minitest rewriter

### LINKED ISSUE #4426: RSpec `describe` nested inside a module triggers error 5067
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0A%23%20Works%0Adescribe%20SomeModule%3A%3ASomeClass%20do%0Aend%0A%0A%23%20Doesn't%20work%0Amodule%20SomeOtherModule%0A%20%20describe%20SomeClass%20do%0A%20%20end%0Aend%0A)

```ruby
# typed: true

# Works
describe SomeModule::SomeClass do
end

# Doesn't work
module SomeOtherModule
  describe SomeClass do
  end
end
```

#### Observed output

```
editor.rb:9: The super class SomeOtherModule of SomeOtherModule::<describe 'SomeClass'> does not derive from Class https://srb.help/5067
     9 |  describe SomeClass do
    10 |  end
Errors: 1
```

#### Expected behavior

Error 5067 is not triggered.

- - -

### PR DESCRIPTION
**Pull request by elliottt**
Only emit a superclass declaration for classes emitted by the minitest DSL when the context is known to be a class.

### Motivation

#4360 introduced a new error when the superclass of a class is a module, but this breaks tests that are written inside of a module as mentioned in #4426.

Fixes #4426 

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_K40yOJYtjPHgN4
→ https://go/builds/bui_K40zqZAeFopDm4

File: rewriter/Minitest.cc [hunk 1]
```diff
@@ -280,43 +280,49 @@ ast::ExpressionPtr runSingle(core::MutableContext ctx, ast::Send *send) {
...
                                                 ctx.state.enterNameConstant("<describe '" + argString + "'>"));
         return ast::MK::Class(send->loc, send->loc, std::move(name), std::move(ancestors), std::move(rhs));
     } else if (send->fun == core::Names::it()) {
         ConstantMover constantMover;
         block->body = ast::TreeMap::apply(ctx, constantMover, move(block->body));
         auto name = ctx.state.enterNameUTF8("<it '" + argString + "'>");
-        auto method = addSigVoid(
-            ast::MK::SyntheticMethod0(send->loc, send->loc, std::move(name), prepareBody(ctx, std::move(block->body))));
+        auto method = addSigVoid(ast::MK::SyntheticMethod0(send->loc, send->loc, std::move(name),
+                                                           prepareBody(ctx, false, std::move(block->body))));
```
Comment (froydnj) [inline rewriter/Minitest.cc:300]:
```suggestion
        const bool bodyIsClass = false;
        auto method = addSigVoid(ast::MK::SyntheticMethod0(send->loc, send->loc, std::move(name),
                                                           prepareBody(ctx, bodyIsClass, std::move(block->body))));
```

File: rewriter/Minitest.cc [hunk 2]
```diff
@@ -280,43 +280,49 @@ ast::ExpressionPtr runSingle(core::MutableContext ctx, ast::Send *send) {
...
+
+        // Avoid subclassing the containing context when it's a module, as that will produce an error in typed: false
+        // files
+        if (isClass) {
+            ancestors.emplace_back(ast::MK::Self(arg.loc()));
+        }
+
         ast::ClassDef::RHS_store rhs;
-        rhs.emplace_back(prepareBody(ctx, std::move(block->body)));
+        rhs.emplace_back(prepareBody(ctx, true, std::move(block->body)));
```
Comment (froydnj) [inline rewriter/Minitest.cc:291]:
```suggestion
        const bool bodyIsClass = true;
        rhs.emplace_back(prepareBody(ctx, bodyIsClass, std::move(block->body)));
```

**Review by froydnj — APPROVED** [review]
LGTM.  Suggestions below, take or leave them at your preference.  I don't know if you'd want to add some brief blurb(s) about why we're passing constants at those particular places.

**Comment by elliottt** [comment]
> LGTM. Suggestions below, take or leave them at your preference. I don't know if you'd want to add some brief blurb(s) about why we're passing constants at those particular places.

I like your suggestion to name the booleans, much clearer than any comment I would write :)

**Pull request merged**